solutions/usaco-670.mdx

```diff
@@ -2,89 +2,107 @@
 id: usaco-670
 source: USACO Gold 2016 December
 title: Cow Checklist
-author: Maggie Liu
+author: Maggie Liu, Ryan Chou
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_checklist_gold_dec16.html)
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_checklist_gold_dec16.html)
 
+Note that at any single point in time Farmer John will be at either a Holstein or a Guernsey, and will have the choice to either go to the next Holstein or Guernsey.
+
+This suggests a DP. Let $\texttt{dp}_{i,j}$ represent the minimum cumulative distance for Farmer John to visit the first $i$ Holsteins and the first $j$ Guernseys.
+<Asterisk>We'll have to add an extra dimension when we're actually implementing the problem to account for which one Farmer John is currently at.</Asterisk>
+
+Then, Farmer John can either go to the next Holstein or the next Guernsey.
+
+$\texttt{dp}_{i, j} = \min(\texttt{dp}_{i - 1,j}, \texttt{dp}_{i,j - 1})$
+
+
+## Implementation
 **Time Complexity:** $\mathcal{O}(H \cdot G)$
 
 <LanguageSection>
-
 <CPPSection>
 
 ```cpp
 #include <iostream>
 #include <cstdio>
+#include <vector>
 using namespace std;
 
-int energy(pair<int, int> a, pair<int, int> b)
-{
-	int dx = a.first - b.first, dy = a.second - b.second;
-	return dx * dx + dy * dy;
+using ll = long long;
+
+struct coord {
+	int x, y;
+};
+
+ll dist(coord &a, coord &b) {
```

Comment:
add `const`

Reply:
wot

Reply:
yeah, but it's not like the function needs a copy or anything